File: unity/Runtime/UnityAds/IUnityAds.cs
namespace EE {
    public interface IUnityAds : IPlugin {
        void Initialize(string gameId, bool testModeEnabled);
        void SetDebugModeEnabled(bool enabled);
        IInterstitialAd CreateInterstitialAd(string adId);
        IRewardedAd CreateRewardedAd(string adId);
    }
}

File: src/ts/src/unity_ads/IUnityAds.ts
import { IFullScreenAd } from "../ads";
import { IPlugin } from "../core";

export interface IUnityAds extends IPlugin {
    initialize(gameId: string, testModeEnabled: boolean): Promise<boolean>;
    setDebugModeEnabled(enabled: boolean): void;
    createInterstitialAd(adId: string): IFullScreenAd;
    createRewardedAd(adId: string): IFullScreenAd;
}